### ISSUE TITLE
remove dependency on window so pusher-js can be imported in workers/shared_workers

### ISSUE THREAD

**Issue by abhishiv** [pusher/pusher-js#499]
Right now in workers i have to externally importScript from cdn instead of pusher being a part of my bundle because it throws `window is not defined`.

```
importScripts("https://js.pusher.com/7.0/pusher.worker.min.js");
```
v/s
```
import Pusher from 'pusher-js'
```

If I import in my bundle `pusher-js`, this following error is thrown 'window is not defined' - I guess it could be easily fixed by modifying webpack config

```
(function webpackUniversalModuleDefinition(root, factory) {
	if(typeof exports === 'object' && typeof module === 'object')
		module.exports = factory();
	else if(typeof define === 'function' && define.amd)
		define([], factory);
	else if(typeof exports === 'object')
		exports["Pusher"] = factory();
	else
		root["Pusher"] = factory();
})(window, function() {
```

Also As noted by @ hph in #183 "window is not defined" error when using webpack + pusher-js + react" the solution that issue was closed with isn't ideal even for SSR, and won't work for use cases like mine.

> Excellent, in that case I'll consider this issue closed for now, although I think we should try to find a better solution.

**Comment by leesio** [comment]
Thanks for reporting this issue.

I think the issue is that:

```
import Pusher from 'pusher-js'
```

Will use the build of pusher-js for the web runtime, rather than the worker runtime. In order to resolve this, we'll need to provide an entrypoint for the worker runtime, similar to how we expose the `pusher-with-encryption` bundle

**Comment by leesio** [comment]
I just released version 7.0.3 which should allow you to import the worker like this:

```
import Pusher from 'pusher-js/worker```

or 

``` 
import Pusher from 'pusher-js/worker/with-encryption'
```

I believe this should solve your issue. If it doesn't, please leave a comment and I'll reopen the issue

**Comment by van-datadive** [comment]
Pusher worker package still access window in `randomInt` function
